src/commands/lambda/functions/instrument.ts

```diff
@@ -134,17 +146,49 @@ export const calculateUpdateRequest = async (
   }
   let needsUpdate = false
 
-  // Update Handler
-  const expectedHandler = HANDLER_LOCATION[runtime]
-  if (config.Handler !== expectedHandler) {
-    needsUpdate = true
-    updateRequest.Handler = HANDLER_LOCATION[runtime]
+  // If a layer version is set for Java or a custom runtime.
+  // We need to inform the customer to only set the extension argument.
+  if (RUNTIME_LOOKUP[runtime] === RuntimeType.JAVA || RUNTIME_LOOKUP[runtime] === RuntimeType.CUSTOM) {
+    if (settings.layerVersion !== undefined) {
+      throw new Error(
+        `Only the extension version should be set for the ${runtime} runtime. Please remove the layer version from the instrument command and only use the extension version.`
+      )
+    }
+  }
+  // Ruby requires handler redirection, which needs to be performed manually.
+  // We need to inform the customer to only set the extension and remove they layer version.
+  if (RUNTIME_LOOKUP[runtime] === RuntimeType.RUBY) {
+    if (settings.layerVersion !== undefined) {
+      throw new Error(
+        'The Ruby layer requires extensive manual instrumentation. Please only set the extension version with the instrument command.'
```

Comment:
I would avoid the word "extensive" here -- that makes it sound scarier than it is 🙂